apps/wallet/src/ui/app/hooks/useGetCoinBalance.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiAddress } from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+
+import { useRpc } from '_hooks';
+type GetCoinBalanceProps = {
+    address?: SuiAddress | null;
+    coinType: string;
+};
+export function useGetCoinBalance({ address, coinType }: GetCoinBalanceProps) {
```

Comment:
Seems like you could probably get away with positional args here but not a deal breaker.